draft-ietf-dnsop-svcb-https.md

```diff
@@ -736,20 +720,31 @@ support, and this informs the underlying transport protocol used (such
 as QUIC-over-UDP or TLS-over-TCP).
 
 ALPNs are identified by their registered "Identification Sequence"
-(alpn-id), which is a sequence of 1-255 octets.
+(`alpn-id`), which is a sequence of 1-255 octets.
+
+    alpn-id = 1*255OCTET
+
+The presentation `value` of "alpn" is a comma-separated list of one or
+more `alpn-id`s, each represented as a character-string:
+
+    alpn-value = char-string *("," char-string)
+
+To decode an `alpn-value`, the parser MUST split the `value` at each ","
```

Comment:
Why doesn't this just say "unescaped comma"?  \044 really does not buy anything over "\," when parsing.   So is '\,' now also a separator or is it a syntax error?

Reply:
The proposed algorithm here is "character-string decode, split on commas, character-string decode again".  Therefore:
* `alpn=ex1,ex2\,ex3` is `ex1`, `ex2`, and  `ex3`, because the first-pass decoding unescapes the entire value
* `alpn=ex1,ex2\\,ex3` is presumably a syntax error, because `example2\` is not a valid character-string, because `\` is supposed to escape something but there is nothing to escape.
* `alpn=ex1,ex2\\\\,ex3` is `ex1`, `ex2\`, and `ex3`

Reply:
If you do this then "foo\044bar" becomes "foo,bar"  which becomes "foo" and "bar".   Following your method you would have to have "foo\\\\044bar" which becomes "foo\044bar" which becomes "foo,bar".   And to get a alpn-id that is "foo\044bar" I would need to enter "foo\\\\\\\\044bar"  (4 backslashes) instead of "foo\\\\044bar" (2 backslashes).

All this for not extending the existing character string parser to know if comma is a special or not.  'if (c == ',' && !escaped) end-of-alpn-id;".